examples/bigquery-analyze-realtime-reddit-data/terraform/main.tf

```diff
@@ -0,0 +1,284 @@
+# service account for reddit VM
+resource "google_service_account" "reddit_service_account" {
+  account_id   = "${var.service_account_name}"
+  display_name = "Reddit VM Service Account"
+}
+
+resource "google_project_iam_binding" "reddit_vm_compute" {
+  project = "${var.project_id}"
+  role    = "roles/compute.admin"
+  members = [
+    "serviceAccount:${google_service_account.reddit_service_account.email}"
+  ]
+}
+
+resource "google_project_iam_binding" "reddit_vm_dataflow" {
+  project = "${var.project_id}"
+  role    = "roles/dataflow.admin"
+  members = [
+    "serviceAccount:${google_service_account.reddit_service_account.email}"
+  ]
+}
+
+resource "google_project_iam_binding" "reddit_vm_pubsub" {
+  project = "${var.project_id}"
+  role    = "roles/pubsub.admin"
+  members = [
+    "serviceAccount:${google_service_account.reddit_service_account.email}"
+  ]
+}
+
+
+resource "google_compute_instance" "reddit_vm" {
```

Comment:
You have mixed project referencing using project from variables or implicitly via global gcloud config, that's a little error prone

Reply:
That's also an issue at other places